ts/nni_manager/core/experimentsManager.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+import { getLogger, Logger } from '../common/log';
+import { isAlive, withLock, withLockSync, getExperimentsInfoPath } from '../common/utils';
+import { ExpManager } from '../common/expmanager';
+import { Deferred } from 'ts-deferred';
+
+interface CrashedInfo {
+    experimentId: string;
+    isCrashed: boolean;
+}
+
+interface FileInfo {
+    buffer: Buffer;
+    mtime: number;
+}
+
+class ExperimentsManager implements ExpManager {
```

Comment:
I would prefer `NNIExperimentsManager || ExperimentsManagerImpl`  implements `ExperimentsManager` instead of `ExperimentsManager` implements `ExpManager`

Reply:
Good suggestion, I will modify them.